lib/charms/sdcore_nrf/v0/fiveg_nrf.py

```diff
@@ -0,0 +1,308 @@
+NRF information and another charm requiring this information.
+
+## Getting Started
+From a charm directory, fetch the library using `charmcraft`:
+
+```shell
+charmcraft fetch-lib charms.sdcore_nrf_operator.v0.fiveg_nrf
+```
+
+Add the following libraries to the charm's `requirements.txt` file:
+- pydantic
+- pytest-interface-tester
+
+### Requirer charm
+The requirer charm is the one requiring the NRF information.
+
+Example:
+```python
+
+from ops.charm import CharmBase
+from ops.main import main
+
+from charms.sdcore_nrf.v0.fiveg_nrf import NRFAvailableEvent, NRFRequires
+
+logger = logging.getLogger(__name__)
+
+
+class DummyFiveGNRFRequirerCharm(CharmBase):
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.nrf_requirer = NRFRequires(self, "fiveg-nrf")
+        self.framework.observe(self.nrf_requirer.on.nrf_available, self._on_nrf_available)
+
+    def _on_nrf_available(self, event: NRFAvailableEvent):
+        nrf_url = event.url
+        <do something with the nrf_url>
+
+
+if __name__ == "__main__":
+    main(DummyFiveGNRFRequirerCharm)
+```
+
+### Provider charm
+The provider charm is the one providing the information about the NRF.
+
+Example:
+```python
+
+from ops.charm import CharmBase, RelationJoinedEvent
+from ops.main import main
+
+from charms.sdcore_nrf.v0.fiveg_nrf import NRFProvides
+
+
+class DummyFiveGNRFProviderCharm(CharmBase):
+
+    NRF_URL = "https://nrf.example.com"
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.nrf_provider = NRFProvides(self, "fiveg-nrf")
+        self.framework.observe(
+            self.on.fiveg_nrf_relation_joined, self._on_fiveg_nrf_relation_joined
+        )
+
+    def _on_fiveg_nrf_relation_joined(self, event: RelationJoinedEvent):
+        relation_id = event.relation.id
+        self.nrf_provider.set_nrf_information(
+            url=self.NRF_URL,
+            relation_id=relation_id,
+        )
+
+    def _on_nrf_url_changed(
+        self,
+    ):
+        self.nrf_provider.set_nrf_information_in_all_relations(
+            url="https://different.nrf.com",
+        )
+
+
+if __name__ == "__main__":
+    main(DummyFiveGNRFProviderCharm)
+```
+
+"""
+
+import logging
+from typing import Optional, Union
+
+from interface_tester.schema_base import DataBagSchema  # type: ignore[import]
+from ops.charm import CharmBase, CharmEvents, RelationChangedEvent
+from ops.framework import EventBase, EventSource, Handle, Object
+from ops.model import Relation
+from pydantic import AnyHttpUrl, BaseModel, Field, ValidationError
+
+# The unique Charmhub library identifier, never change it
+LIBID = "cd132a12c2b34243bfd2bae8d08c32d6"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+PYDEPS = [
+    "pydantic" "pytest-interface-tester",
```

Comment:
```suggestion
    "pydantic", "pytest-interface-tester",
```
Missing a comma.

Is the pytest one needed outside of testing?

Reply:
Fixed. I've also moved `pytest-interface-tester` to the `test-requirements.txt`.